src/components/todo/TodoList.js

```diff
@@ -0,0 +1,64 @@
+import { useCallback } from 'react';
+import { TodoItem } from './TodoItem';
+import styled from 'styled-components';
+
+// 할 일 집합
+export default function TodoList({ title, todos, setTodos }) {
+  // 할 일 토글 함수
+  const toggleComplete = useCallback(
+    (id) => {
+      setTodos((prevTodos) =>
+        prevTodos.map((todo) =>
+          todo.id === id ? { ...todo, isCompleted: !todo.isCompleted } : todo
+        )
+      );
+    },
+    [setTodos]
+  );
+
+  // 할 일 삭제
+  const handleDelete = useCallback(
+    (id) => {
+      setTodos((prevTodos) => prevTodos.filter((todo) => todo.id !== id));
+    },
+    [setTodos]
+  );
+
+  return (
+    <StyledList>
+      <div className="statusTitle">
+        <span>{title}</span>
+        <span className="count">{todos.length}</span>
+      </div>
+      <ul>
+        {todos.map((todo) => (
+          <TodoItem
+            {...todo}
+            onDelete={() => handleDelete(todo.id)}
+            toggleComplete={() => toggleComplete(todo.id)}
+          />
+        ))}
```

Comment:
todos 배열을 map 함수로 렌더링할 때, 각 항목에 고유한 key prop을 제공해야 리액트에서 효율적으로 렌더링을 할 수 있다고 해요!

```suggestion
{todos.map((todo) => (
  <TodoItem
    key={todo.id}
    {...todo}
    onDelete={() => handleDelete(todo.id)}
    toggleComplete={() => toggleComplete(todo.id)}
  />
))}

```